### PR TITLE
Add ManagementAPI support to create transaction count report

### PR DESCRIPTION
**Pull request by sajinieKavindya**
## Purpose
Implement the following Management APIs for the transaction count component.

| API  | Description | Response |
| ------------- | ------------- | ------------- | 
| /management/transactions/count  | Get the transaction count for the current month | {"Year" : 2020, "Month" : 05, "TransactionCount" : 239347}|
| /management/transactions/count?year=[year]&month=[month]  | Get the transaction count for the given date | {"Year" : 2020, "Month" : 06, "TransactionCount" : 1232347}|
| /management/transactions/report?start=[start]&end=[end]  | Get the report content for the period between start and end date. Generate a report inside <CARBON_HOME>/temp directory  | {"TransactionCountData" : [[date, nodeId, count, encryptedCount], [val1, val2, val3, val4]]}|
| /management/transactions/report?start=[start]  | Get the report content for the period between start and current date. Generate a report inside <CARBON_HOME>/temp directory  | {"TransactionCountData" : [[date, nodeId, count, encryptedCount], [val1, val2, val3, val4]]}|

### CODE REVIEW

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/RequestCountResource.java [hunk 1]
```diff
@@ -61,26 +70,48 @@
...
+            String monthParameter = Utils.getQueryParameter(synCtx, "month");
+
+            return handleTransactionCountCommand(axis2MessageContext, yearParameter, monthParameter);
+
+        } else if ("report".equalsIgnoreCase(pathParam)) {
+            String start = Utils.getQueryParameter(synCtx, "start");
+            String end = Utils.getQueryParameter(synCtx, "end");
+
+            return handleTransactionReportCommand(axis2MessageContext, start, end);
+
```
Comment (sachithKay) [inline components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/RequestCountResource.java:87]: lets remove extra lines.

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/RequestCountResource.java [hunk 2]
```diff
@@ -92,41 +123,154 @@ public boolean invoke(MessageContext synCtx,
...
+        JSONObject response;
+        axisCtx.setProperty(org.apache.axis2.Constants.Configuration.MESSAGE_TYPE, "application/json");
+        axisCtx.setProperty(Constants.Configuration.CONTENT_TYPE, "application/json");
+
+        TransactionStore transactionStore = DataHolder.getInstance().getTransactionStore();
+        if (null == transactionStore) {
+            response = Utils.createJsonError("Transaction Count Handler component is not initialized", axisCtx,
+                                             FORBIDDEN);
+            Utils.setJsonPayLoad(axisCtx, response);
+            return true;
```
Comment (sachithKay) [inline components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/RequestCountResource.java:210]: L 231, 232 already have this right? No need to repeat here right?

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/Constants.java [hunk 3]
```diff
@@ -43,7 +43,8 @@
     public static final String PREFIX_LOGOUT = "/logout";
     public static final String PREFIX_SERVER_DATA = "/server";
     public static final String PREFIX_LOG_FILES = "/logs";
-    public static final String PREFIX_REQ_COUNT = "/transactions";
+    public static final String PREFIX_TRANSACTION = "/transactions";
+    public static final String PATH_PARAM_TRANSACTION = "/" + "{param}";
```
Comment (sdkottegoda) [inline components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/Constants.java:47]: which param are we expecting here?

**Comment by sachithKay** [comment]
Let's refactor the code upon merging.

**Pull request merged**